src/index.js

```diff
@@ -45,3 +45,22 @@ exports.Queue = Plotly.Queue;
 
 // export d3 used in the bundle
 exports.d3 = require('d3');
+
+Plotly.register({
+    traces: {
+        bar: require('./traces/bar'),
```

Comment:
Nice!

Bringing @bpostlethwaite and @alexcjohnson for some thoughts on the API:

- I'm thinking that making `Plotly.register` take an option object might not be necessary.

```js
// this might be sufficient
Plotly.register([require('plotly.js/bar'), require('plotly.js/box')]);

// or even where we would loop over the argument object
Plotly.register(require('plotly.js/bar'), require('plotly.js/box'));
```

Granted, passing an object with key `traces` makes the API easily extendible to registering components down the road. But,  do users need to know that they are registering trace or component modules? Could we handle the trace / component split behind the scene inside the `register` function (e.g. by adding a *type* property to each trace module)?